Winget-AutoUpdate-Install.ps1

```diff
@@ -538,8 +538,14 @@ Write-Host "`t________________________________________________________`n `n "
 if (!$Uninstall) {
     Write-ToLog "Installing WAU to $WAUinstallPath\"
     Install-Prerequisites
-    Update-Winget
-    Install-WingetAutoUpdate
+    $UpdateWinget = Update-Winget
+    if ($UpdateWinget -ne "fail") {
+        Write-Host "`r" #Extra Line in console only
+        Install-WingetAutoUpdate
+    }
+    else {
+        Write-ToLog "Winget is mandatory to execute WAU." "Red"
+    }

```

Comment:
But, if WinGet is installed (older version) and the upgrade fails then it WinGet still exist but WAU won't be installed (or am I wrong?)...

Reply:
hmm, that's true...
but if Winget is not up to date we've seen that it can cause issues.
We have 2 options:
- Roll back my change
- precise the log to `Write-ToLog "Latest Winget Version is mandatory to execute WAU." "Red"`

Reply:
Then precise the log (I haven't seen it happen yet).

Reply:
...and `Update-StoreApps` runs also, so that can trigger a successed install when trying to install WAU again

Reply:
One could check the local WinGet version again after `Update-StoreApps`

Reply:
rolled back in my dev branch
https://github.com/Romanitho/Winget-AutoUpdate/tree/draft-upgrade

Reply:
and added a workaround when API rate limit is reach:
https://github.com/Romanitho/Winget-AutoUpdate/commit/f01a7ccefb8f963ee3f8a9dad51f2fa4249e7205

Reply:
Ah, nice catch 🥳
And if the client is healthy `Update-StoreApps` upgrades Winget too.